packages/liveblocks-react-ui/src/primitives/Composer/utils.ts

```diff
@@ -175,19 +200,125 @@ export function commentBodyToComposerBody(body: CommentBody): ComposerBody {
     .filter(exists);
 }
 
-export function getPlacementFromPosition(
-  position: SuggestionsPosition,
-  direction: Direction = "ltr"
-): Placement {
-  return `${position}-${direction === "rtl" ? "end" : "start"}`;
+export function getRtlFloatingAlignment(
+  alignment: FloatingAlignment
+): FloatingAlignment {
+  switch (alignment) {
+    case "start":
+      return "end";
+    case "end":
+      return "start";
+    default:
+      return "center";
+  }
 }
 
-export function getSideAndAlignFromPlacement(placement: Placement) {
+export function getSideAndAlignFromFloatingPlacement(placement: Placement) {
   const [side, align = "center"] = placement.split("-");
 
   return [side, align] as const;
 }
 
+export function useOnComposerEditorChangeWithEventSource(
+  editorChangeEventSource: EventSource<void>,
+  callback: () => void
+) {
```

Comment:
Is this hook exported purely because we want to also listen to editor `onchange` events inside `Composer.Form`?

Reply:
In the `Composer.Form` parent but also in children components (e.g. `ComposerEditorMentionSuggestionsWrapper`, `ComposerEditorFloatingToolbarWrapper`, etc).